### PR TITLE
Pass in a reason to all cancel calls

### PR DESCRIPTION
**Pull request by ryanwalls**

### CODE REVIEW

File: workflow/client.go [hunk 1]
```diff
@@ -234,15 +234,17 @@ func (c *client) CompleteSuccessfulActivity(workflowID, activityID string, resul
...
+func (c *client) CompleteCancelledActivity(workflowID, activityID, reason, details string) (*models.Activity, error) {
 	token, err := c.tokenFetcher.Token(c.audience)
 	if err != nil {
 		return nil, err
 	}
 	cancelledActivity := &models.Activity{
 		ID:     swag.String(activityID),
 		Status: swag.String(models.ActivityStatusCancelled),
-		Error:  &models.ActivityError{Details: details},
+		Error:  &models.ActivityError{Reason: swag.String(reason), Details: details},
```
Comment (pkrull) [inline workflow/client.go:247]: Formatting? Maybe Details on another line to be consistent.

File: workflow/client_test.go [hunk 2]
```diff
@@ -595,7 +595,7 @@ func TestCompleteCancelledActivity(t *testing.T) {
 		expectedActivity := &models.Activity{
 			ID:     swag.String(activityID),
 			Status: swag.String(models.ActivityStatusCancelled),
-			Error:  &models.ActivityError{Details: "some cancel details"},
+			Error:  &models.ActivityError{Reason: swag.String("some reason"), Details: "some cancel details"},
```
Comment (pkrull) [inline workflow/client_test.go:598]: Formatting?

**Review by pkrull — APPROVED** [review]

File: workflow/client_test.go (hunk 2 above)
Comment (ryanwalls) [reply workflow/client_test.go:598]: @pkrull What's wrong with the formatting?

File: workflow/client.go (hunk 1 above)
Comment (ryanwalls) [reply workflow/client.go:247]: Ah, I personally only break initializers across lines if the line is too long.  1 or 2 fields, is okay.  More than that and probably want multiple lines.

File: workflow/client_test.go (hunk 2 above)
Comment (ryanwalls) [reply workflow/client_test.go:598]: Ah, I personally only break initializers across lines if the line is too long.  1 or 2 fields, is okay.  More than that and probably want multiple lines.

**Pull request merged**